<a href="https://colab.research.google.com/github/binodpun/Data-Science-/blob/main/Data_profiling_and_inspection.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
# https://github.com/mwaskom/seaborn-data/blob/master/tips.csv

In [4]:
df = pd.read_csv("/content/tips.csv")

In [7]:
df.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [9]:
df.tail()

total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

In [11]:
df['sex'].value_counts()

sex
Male      157
Female     87
Name: count, dtype: int64

In [16]:
df.dtypes

total_bill    float64
tip           float64
sex            object
smoker         object
day            object
time           object
size            int64
dtype: object

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


In [19]:
df.shape

(244, 7)

* data has 244 rows
* data has 7 columns
* out of 244 rows each column has 244 not-null values which means data is not missing
* two column have float64 datatype
* four column have object datatype
* one column have int64 datatype

In [20]:
df.describe()

total_bill         tip        size
count  244.000000  244.000000  244.000000
mean    19.785943    2.998279    2.569672
std      8.902412    1.383638    0.951100
min      3.070000    1.000000    1.000000
25%     13.347500    2.000000    2.000000
50%     17.795000    2.900000    2.000000
75%     24.127500    3.562500    3.000000
max     50.810000   10.000000    6.000000

total_bill:

* count 244 means data has 244 not-null values
* means: on average people spend 19.78 dollar in hotel
* the standard deviation is 8.90
* minimum total bill in a data 3.07
* 25% means;
     * below 13.34 we have 25% of data point
*50% means;
    * 17.79 total bill vanda muni 50% data cha
* max total bill 50.81 dollars            